### PR TITLE
Add Typescript Sourcemaps

### LINKED ISSUE #506: Link compiled webpack code back to typescript sources
From #455:

> AFAIK there is a way to link it again back to the original TypeScript source, we just have not done it yet.

### PR DESCRIPTION
**Pull request by saulshanabrook**
Create sourcemaps when compiling TS files so that you can see
the original typescript source when you debug the app. This requires
a more resource intensive sourcemap creation in webpack, so we don't
enable it by default. On my laptop, enabling it increased build times
from ~9s to ~13s..

Fixes #506

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -90,6 +90,10 @@ jupyterlab_launcher` to get the latest version.
 
 * To install JupyterLab in isolation for a single conda/virtual environment, you can add the `--sys-prefix` flag to the extension activation above; this will tie the installation to the `sys.prefix` location of your environment, without writing anything in your user-wide settings area (which are visible to all your envs):
 
+* You can run `npm run build:main -- -- -- --devtool source-map` to build more accurate sourcemaps that show the original
```
Comment (blink1073) [inline CONTRIBUTING.md]: Are the extra `--` needed here?

Comment (saulshanabrook) [reply CONTRIBUTING.md]: @blink1073 Yeah, because it calls `npm run` many times nested so it needs that many to get to the `webpack` command.

Comment (saulshanabrook) [reply CONTRIBUTING.md]: @blink1073 created some more npm run command instead. How does it look now?

**Comment by blink1073** [comment]
This is great, thanks!

**Pull request merged**